.github/workflows/codeql.yml

```diff
@@ -0,0 +1,45 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+  schedule:
+    - cron: '22 9 * * 2'
+
+env:
+  WEECHAT_DEPENDENCIES: devscripts equivs python3-pip cmake ninja-build lcov pkg-config libncurses-dev gem2deb libperl-dev python3-dev libaspell-dev liblua5.3-dev tcl8.6-dev guile-3.0-dev libv8-dev libcurl4-gnutls-dev libgcrypt20-dev libgnutls28-dev libzstd-dev zlib1g-dev curl libcpputest-dev php-dev libphp-embed libargon2-dev libsodium-dev flake8 pylint python3-bandit asciidoctor ruby-pygments.rb shellcheck
```

Comment:
Thanks for the PR, there's a codeql branch, I made an attempt 2 years ago but it was not working.

I wonder if these dependencies can be somewhat put in a file included in both ci.yml and codeql.yml?
I don't know if we can include yaml files here.

Reply:
>I wonder if these dependencies can be somewhat put in a file included in both ci.yml and codeql.yml?
I don't know if we can include yaml files here.

I'm sure there's a way; Actions is quite flexible.

But if the goal is to just have one dependency list, then perhaps the simplest solution would be to merge the codeql.yml and ci.yml file together into one file with 2 jobs? Then they could both reference the same environment variable.